File: old/aerox.Runtime.Widgets/Events/ScrollEvent.cs
using aerox.Runtime.Math;

namespace aerox.Runtime.Widgets.Events;

public class ScrollEvent : Event
{
    public Vector2<float> Delta;
    public Vector2<float> Position;

    public ScrollEvent(WidgetSurface surface, Vector2<float> position, Vector2<float> delta) : base(surface)
    {
        Position = position;
        Delta = delta;
    }
}

File: modules/widgets/lib/aerox/widgets/event/ScrollEvent.cpp
#include "aerox/widgets/event/ScrollEvent.hpp"

namespace aerox::widgets
{
    ScrollEvent::ScrollEvent(const Shared<Surface>& inSurface, const Vec2<float>& inDelta,
        const Vec2<float>& inPosition) : Event(inSurface), delta(inDelta), position(inPosition)
    {
    }
}
